src/communication/channel.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "channel.hpp"
+#include "log.hpp"
+#include "utils/aosheader.hpp"
+#include "utils/checksum.hpp"
+
+namespace aos::zephyr::communication {
+
+Mutex Channel::sWriteMutex;
+
+Channel::Channel(TransportItf* transport, int port)
+    : mTransport(transport)
+    , mPort(port)
+{
+}
+
+Error Channel::Connect()
+{
+    UniqueLock lock(mMutex);
+
+    LOG_DBG() << "Connect channel port=" << mPort;
+
+    if (mTransport->IsOpened()) {
+        return ErrorEnum::eNone;
+    }
+
+    return ErrorEnum::eRuntime;
+}
+
+Error Channel::Close()
+{
+    UniqueLock lock(mMutex);
+
+    LOG_DBG() << "Close channel port=" << mPort;
+
+    mClose = true;
+    mCondVar.NotifyAll();
+
+    return ErrorEnum::eNone;
+}
+
+int Channel::Read(void* buffer, size_t size)
```

Comment:
comments say that data should be read from channel. I expected data to be read from transport in the input data

Reply:
But we read and write from channel, because when we call this method we read something from channel, and we read from transport but using for that like subscribe approach